### PR TITLE
CXSPA-4364 Fix B2B user creation UI issue:

### PR DESCRIPTION
**Pull request by rmch91**
Closes: https://jira.tools.sap/browse/CXSPA-4364

### CODE REVIEW

File: feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts [hunk 1]
```diff
@@ -629,12 +640,34 @@ export class B2BUserEffects {
...
+  );
   constructor(
     private actions$: Actions,
     private b2bUserConnector: B2BUserConnector,
     private routingService: RoutingService,
     private userAccountFacade: UserAccountFacade,
-    private userIdService: UserIdService
+    private userIdService: UserIdService,
+    @Optional()
+    private b2BUserCreationNotifierService?: B2BUserCreationNotifierService
```
Comment (Matejk00) [inline feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts]: I think we could make this `b2BUserCreationNotifierService?` service as `protected`. This is recommended for the better extensibility.

File: feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts [hunk 2]
```diff
@@ -629,12 +640,34 @@ export class B2BUserEffects {
     )
   );
 
+  // TODO: make b2BUserCreationNotifierService a required dependency
```
Comment (Matejk00) [inline feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts]: Please create a Jira ticket for it, and but the number into the code comment.
https://sap.github.io/spartacus-docs/breaking-changes/#adding-new-constructor-dependencies-in-minor-versions

File: feature-libs/organization/administration/core/services/b2b-user-creation-notifier.service.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { B2BUser } from '@spartacus/core';
+import { Observable, Subject } from 'rxjs';
+import { LoadStatus } from '../model';
+
+interface UserCreationNotificationData {
```
Comment (Matejk00) [inline feature-libs/organization/administration/core/services/b2b-user-creation-notifier.service.ts]: `UserCreationNotifierData` sounds better to me. But it's up to you. :)

File: feature-libs/organization/administration/components/user/services/user-item.service.ts [hunk 4]
```diff
@@ -19,11 +20,31 @@ import { CurrentUserService } from './current-user.service';
   providedIn: 'root',
 })
 export class UserItemService extends ItemService<B2BUser> {
+  // TODO: make b2BUserCreationNotifierService a required dependency
```
Comment (Matejk00) [inline feature-libs/organization/administration/components/user/services/user-item.service.ts]: Please create a Jira ticket for it, and but the number into the code comment.
https://sap.github.io/spartacus-docs/breaking-changes/#adding-new-constructor-dependencies-in-minor-versions

**Review by Matejk00 — CHANGES_REQUESTED** [review]

File: feature-libs/organization/administration/core/services/b2b-user-creation-notifier.service.ts (hunk 3 above)
Comment (Matejk00) [reply feature-libs/organization/administration/core/services/b2b-user-creation-notifier.service.ts]: Couldn't be also this part as a part of the model file?

File: feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts (hunk 1 above)
Comment (rmch91) [reply feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts]: Changed

File: feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts (hunk 2 above)
Comment (rmch91) [reply feature-libs/organization/administration/core/store/effects/b2b-user.effect.ts]: Changed

File: feature-libs/organization/administration/core/services/b2b-user-creation-notifier.service.ts (hunk 3 above)
Comment (rmch91) [reply feature-libs/organization/administration/core/services/b2b-user-creation-notifier.service.ts]: Changed

File: feature-libs/organization/administration/components/user/services/user-item.service.ts (hunk 4 above)
Comment (rmch91) [reply feature-libs/organization/administration/components/user/services/user-item.service.ts]: Changed

**Review by Matejk00 — APPROVED** [review]

**Pull request merged**